File: backend/src/main/java/com/plugga/backend/dao/UserDeckDAO.java
package com.plugga.backend.dao;

import com.plugga.backend.entity.UserDeck;

import java.util.List;

public interface UserDeckDAO {
    List<UserDeck> findAll();

    UserDeck findById(int id);

    List<UserDeck> findByUserId(int userId);

    List<UserDeck> findByDeckId(int deckId);

    void save(UserDeck userDeck);

    void deleteById(int id);
}

File: package/src/main/kotlin/com/plugga/backend/dao/UserDeckDAO.kt
package com.plugga.backend.dao

import com.plugga.backend.entity.UserDeck

interface UserDeckDAO {
    fun findAll(): MutableList<UserDeck>

    fun findById(id: Int): UserDeck?

    fun findByUserId(userId: Int): MutableList<UserDeck>

    fun findByDeckId(deckId: Int): MutableList<UserDeck>

    fun save(userDeck: UserDeck)

    fun deleteById(id: Int)
}
